back-nest/src/cart/cart.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, Logger, UnauthorizedException } from '@nestjs/common';
+
+import { PrismaService } from '../../prisma/prisma.service';
+import { UpdateCartParams } from './model/updateCartParams.model';
+
+@Injectable()
+export class CartService {
+  private readonly logger = new Logger(CartService.name);
+
+  constructor(private prisma: PrismaService) {}
+
+  async getUserCart(user_id: number) {
+    if (!user_id) {
+      throw new UnauthorizedException('Unauthorized');
+    }
```

Comment:
You don't need to check the user_id here. Your middleware must do this job instead. So please remove it